libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/LocationEx.kt

```diff
@@ -116,12 +117,21 @@ internal fun Bundle.toMap(): FixLocationExtras {
     while (iterator.hasNext()) {
         val key = iterator.next()
         when (val value = this.get(key)) {
+            is Boolean -> map[key] = Value(value)
+            is Byte -> map[key] = Value(value.toLong())
+            is Char -> map[key] = Value(value.toString())
             is Double -> map[key] = Value(value)
+            is Float -> map[key] = Value(value.toDouble())
+            is Int -> map[key] = Value(value.toLong())
             is Long -> map[key] = Value(value)
-            is Boolean -> map[key] = Value(value)
+            is Short -> map[key] = Value(value.toLong())
             is String -> map[key] = Value(value)
             else -> {
-                // do nothing
+                logW(
+                    TAG,
+                    "Unsupported type in location extras " +
+                        "`${value?.javaClass?.kotlin?.simpleName}`"
```

Comment:
This may be obfuscated when using Proguard. Could we obtain the type differently? Do we even need to log it? 🤔

Reply:
> Could we obtain the type differently?

How can we do that?

> Do we even need to log it?

I have no strong opinion here and can just remove that, but I thought it can be useful while trying to understand what is wrong.